improver/blending/weighted_blend.py

```diff
@@ -721,13 +718,31 @@ def weighted_mean(self, cube, weights):
         """
         weights_array = self.non_percentile_weights(cube, weights)
 
-        # Calculate the weighted average.
-        cube_new = collapsed(
-            cube, self.blend_coord, iris.analysis.MEAN, weights=weights_array
+        (collapse_dim,) = cube.coord_dims(self.blend_coord)
+        slice_dim = 1 if collapse_dim == 0 else 0
+
+        slicing_threshold = (
+            3e7  # This allows for arrays ~100MB and smaller not to be sliced
+        )
+        allow_slicing = (
+            cube.ndim > 2 and np.array(cube.shape).prod() > slicing_threshold
```

Comment:
Why `cube.ndim > 2`?  I think this might actually need to be `cube.ndim > 3`.

Here's my logic: in the majority of cases (actually all envisaged cases) IMPROVER will be dealing with data that has a spatial x- and y-dimension.  That accounts for two dimensions.  Add on to this a blend coord to collapse over, and your slice coord needs to be different.  For example: if you are blending probabilities (which we usually are), there are two cases:
1) Cycle blending: blend dimension is forecast reference time, slice dimension is threshold
2) Multi-model blending: blend dimension is model ID, slice dimension is threshold

In each case there are 4 dimensions.  The other case where you might be doing this is where it's called by the `threshold` plugin / CLI.  In that case, the blend dimension is realization, and the slice dimension is threshold - again, 4 dimensions.

The only case where you wouldn't need this to be `cube.ndim > 3` is if we were blending spot data, which has only 1 "spatial" axis (a spot location).  However the IMPROVER system is designed to do everything on the grid and extract at level 3 - so we will not be doing that - and even if spot data were blended it would be unlikely to meet the array size criterion for requiring slicing.

Reply:
I'm happy with `cube.ndim > 3` here and then we can dispense with the `slicing_threshold` altogether (cube slices in threshold will be 3D). 

I'm just not entirely happy with having this condition there in the first place, though. As I wrote above https://github.com/metoppv/improver/pull/1260#pullrequestreview-416439999, it might be better to leave this decision to the caller rather than to have the best guess logic here that may not work well in some cases. Weighted mean is a fairly generic operation that may pop out in other contexts (threshold cli is a good example).